src/main/java/com/ncc/aif/ValidateAIFCli.java

```diff
@@ -0,0 +1,597 @@
+            } catch (IOException ioe) {
+                logger.warn("---> Could not write validation report for " + fileToValidate + ".");
+            }
+            logger.info("--> Saved validation report to " + outputFilename);
+        }
+
+        return validationReport.getModel().listStatements(null, SH.resultSeverity, SH.Violation).toList().size();
+    }
+
+    // Return false if file is > 5MB or size couldn't be determined, otherwise true
+    private static boolean checkHypothesisSize(File fileToValidate) {
+        try {
+            final Path path = Paths.get(fileToValidate.toURI());
+            final long fileSize = Files.size(path);
+            if (fileSize > 1024 * 1024 * 5) { // 5MB
+                logger.warn("---> Hypothesis KB " + fileToValidate + " is more than 5MB (" + fileSize + " bytes); skipping.");
+                return false;
+            } else {
+                return true;
+            }
+        } catch (IOException ioe) {
+            logger.warn("---> Could not determine size for hypothesis KB " + fileToValidate + "; skipping.");
+            return false;
+        }
+    }
+
+    // Display a summary to the user
+    private static ReturnCode displaySummary(int fileCount, int invalidCount, int skipCount, int abortCount) {
+        final int validCount = fileCount - invalidCount - skipCount;
+        logger.info("Summary:");
+        logger.info("\tFiles submitted: " + fileCount);
+        logger.info("\tSkipped files: " + skipCount);
+        logger.info("\tKBs sent to validator: " + (fileCount - skipCount));
+        logger.info("\tValid KBs: " + (fileCount - invalidCount - skipCount));
+        logger.info("\tInvalid KBs: " + invalidCount);
+        if (abortCount > 0) {
+            logger.info("\t  Aborted validations: " + abortCount);
+        }
+        if (fileCount == validCount) {
+            logger.info("*** All submitted KBs were valid. ***");
+        } else if (fileCount == skipCount) {
+            logger.info("*** No validation was performed. ***");
+        }
+
+        if (invalidCount > 0) { // Return a failure code if anything fails to validate.
+            return ReturnCode.VALIDATION_ERROR;
+        } else {
+            return skipCount == 0 ? ReturnCode.SUCCESS : ReturnCode.FILE_ERROR;
+        }
+    }
+
+    /**
+     * A statistics collector for use in profiling TopBraid-based SHACL validation.  Typical usage is to call
+     * {@link #startCollection()} and {@link #endCollection()} to bound statistics collection,
+     * then call {@link #dump(String)} to dump slow query statistics to <filename>-stats.txt.
+     * <p>
+     * This statistics collector only outputs slow queries via the {@link #dump(String)} method.  If you suspect
+     * validation will not complete due to out of memory or other error conditions, consider using {@link ProgressiveStatsCollector}.
+     */
+    private static class StatsCollector {
```

Comment:
This seems like a good time to move the profilers to a separate Java file.  WDYT?

Reply:
I would have, but I'm waiting for direction on package structure. Should there be a validator package?

Reply:
Based on discussion, we're deferring this.